LukasWaveformReaderApp/src/WaveformReader.cpp

```diff
@@ -79,6 +79,108 @@ void WaveformReader::statusCheck(void)
   std::cout << "------------------------------------------------------------------------" << std::endl;
 }
 
+void WaveformReader::fft(void)
+{
+  int maxIndex = findMaxIndex();
+  int low, high;
+  findRange(low, high, maxIndex);
+  // Define the length of the complex arrays
+  int n = high - low + 1; // buffer size from waveformConfigure call in st.cmd //200000
+  // Input array
+  // IS OURS A REAL VALUED SIGNAL
+  // Dynamically allocate the array because the size can change
+  fftw_complex* x = new fftw_complex[n]; // This is equivalent to: double x[n][2];
+  // Output array
+  fftw_complex* y = new fftw_complex[n];
+  // Samping frequency
+  double sampling_frequency = 3.57142857143e8; // NEED VALUE OF SAMPLING FREQUENCY
+
+  //std::cout << "Just trying maxIndex: " << findMaxIndex() << std::endl;
+  //std::cout << "Above the first for loop" << std::endl;
+  // Fill the first array with data from waveformData
+  for (int i = 0; i < n; i++)
+  {
+      x[i][REAL] = waveformData[i + low];
+      x[i][IMAG] = 0;
+  }
+  //Plant the FFT and execute it
+  fftw_plan plan = fftw_plan_dft_1d(n, x, y, FFTW_FORWARD, FFTW_ESTIMATE);
+  fftw_execute(plan);
+  //Do some cleaning
+  fftw_destroy_plan(plan);
+  fftw_cleanup();
+  // Display the results
+  std::cout << "FFT = " << std::endl;
+  double frequency;
+  std::cout << "---------------------------------------------------------------------------------------------------------------" << std::endl;
+  std::cout << "|    No.    | FREQUENCY (in Hz) |             FFT RESULT             |     MAGNITUDE     | PHASE (in radians) |" << std::endl;
+  std::cout << "---------------------------------------------------------------------------------------------------------------" << std::endl;
+  for (int i = 0; i < n; i++)
```

Comment:
You might want to give the option to show like only outlier values or something for this method, or maybe outputting to a csv file instead of printing it to the terminal